### PR TITLE
Support enter key for calling onClick on Button

### PR DESCRIPTION
**Pull request by lenahong**
### Platforms Impacted
- [ ] iOS
- [ ] macOS
- [X] win32
- [ ] windows
- [ ] android

### Description of changes

onKeyDown recognizes the enter key press to call the onClick event for Button.

### Verification

Manually tested with Buttons recognizing an enter key press when focus and calling onClick event.

### Pull request checklist

This PR has considered (when applicable):
- [ ] Automated Tests
- [ ] Documentation and examples
- [X] Keyboard Accessibility
- [ ] Voiceover
- [ ] Internationalization and Right-to-left Layouts

### CODE REVIEW

File: packages/components/Button/src/Button.tsx [hunk 1]
```diff
@@ -25,6 +25,14 @@ export const Button = compose<IButtonType>({
     } = userProps;
     // attach the pressable state handlers
     const pressable = useAsPressable({ ...rest, onPress: onClick });
+    const onKeyDown = React.useCallback(
+      e => {
+        if (onClick && e.nativeEvent.key === 'Enter') {
+          onClick();
```
Comment (dotnetjunky) [inline packages/components/Button/src/Button.tsx:31]: If you have handled the event, you need to call `e.stopPropagation()` to stop it from bubbling.

File: change/@fluentui-react-native-adapters-2020-05-26-21-34-00-lehon-keyboardClick.json [hunk 2]
```diff
@@ -0,0 +1,8 @@
+{
```
Comment (dotnetjunky) [inline change/@fluentui-react-native-adapters-2020-05-26-21-34-00-lehon-keyboardClick.json:1]: Why do we need to add change file to unrelated packages like this one? @JasonVMo

**Comment by dotnetjunky** [comment]
You change looks like it impact more than win32. Are you sure it only applies to win32?

**Comment by lenahong** [comment]
> You change looks like it impact more than win32. Are you sure it only applies to win32?



> You change looks like it impact more than win32. Are you sure it only applies to win32?

You're right, but I wasn't sure which platforms support keyboard press events since I'm not able to test on any other than win32 and uwp. On UWP, we don't support keyboard focus yet

File: packages/components/Button/src/Button.tsx [hunk 3]
```diff
@@ -25,6 +25,15 @@ export const Button = compose<IButtonType>({
     } = userProps;
     // attach the pressable state handlers
     const pressable = useAsPressable({ ...rest, onPress: onClick });
+    const onKeyDown = React.useCallback(
+      e => {
+        if (onClick && e.nativeEvent.key === 'Enter') {
```
Comment (NikoAri) [inline packages/components/Button/src/Button.tsx:30]: >Enter [](start = 46, length = 5)

What about space key???

Comment (lenahong) [reply packages/components/Button/src/Button.tsx]: Added!

Comment (FalseLobster) [reply packages/components/Button/src/Button.tsx]: Actually, usually buttons only support space, not enter.  We should check to make sure this is the expected design.

Comment (FalseLobster) [reply packages/components/Button/src/Button.tsx]: Or maybe that's only true for other widgets like Checkboxes?  Disregard :)

Comment (lenahong) [reply packages/components/Button/src/Button.tsx]: I checked buttons on Office apps, and they supported enter!

**Pull request merged**